### ISSUE TITLE
Querying dimensions remotely and modifying remote tensors.

### ISSUE THREAD

**Issue by ratmcu** [OpenMined/PySyft#3163]
I was working on the BERT for an NER classification and my model has a froward pass as below. and my model is already being sent to the remote worker.
```
def forward(self, input_ids, token_type_ids=None):
        seq_length = input_ids.size(1)
        position_ids = torch.arange(seq_length, dtype=torch.long, device=input_ids.device)
        position_ids = position_ids.unsqueeze(0).expand_as(input_ids)
        if token_type_ids is None:
            token_type_ids = torch.zeros_like(input_ids)
        words_embeddings = self.word_embeddings(input_ids)
        position_embeddings = self.position_embeddings(position_ids)
        token_type_embeddings = self.token_type_embeddings(token_type_ids)
        embeddings = words_embeddings + position_embeddings + token_type_embeddings
        embeddings = self.LayerNorm(embeddings)
        embeddings = self.dropout(embeddings)
        return embeddings
```

I cannot create a tensor remotely with the dimensions of another tensor situated in the remote worker without calling a get() on the aforementioned tensor.
eg: in forward pass of the model ``` seq_length = input_ids.size(1)```  here ```seq_length``` cannot be queried without calling a ```get()``` on the ```input_ids``` which seems to violate the total privacy of the data belonging to the remote worker and some performance hit too. 
For now I worked around this by changing dimensions to stay a fixed value. which resulted in a certain loss of convergence speed. 
I thought it'd be a nice to have when dealing with this sort of models.

**Comment by LaRiffle** [comment]
Hey @ratmcu, size() is currently being investigated in https://github.com/OpenMined/PySyft/issues/3382
Meanwhile, I suggest that you use `input_ids.shape[1]` instead it should work correctly!
